### PR TITLE
SOC-5361: use ES for connection page

### PR DESCRIPTION
**Pull request by nttuyen**
optimize a little on query searching in identity and use ES for search in connections instead of JPA query

### CODE REVIEW

File: lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java [hunk 1]
```diff
@@ -254,10 +254,10 @@ private String buildTypeEx(Type type) {
         result = "connections";
         break;
       case INCOMING:
-        result = "incomings";
+        result = "outgoings";
         break;
       case OUTGOING:
-        result = "outgoings";
+        result = "incomings";
```
Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java]: Does not look very logical :) Not a mistake ? Can you explain ?

Also, we should probably have an enum here with a value : CONFIRMED("connections")

File: lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSRelationshipStorageImpl.java [hunk 2]
```diff
@@ -260,33 +260,38 @@ public boolean hasRelationship(Identity identity1, Identity identity2, String re
 
   @Override
   public List<Identity> getConnectionsByFilter(Identity existingIdentity, ProfileFilter profileFilter, long offset, long limit) throws RelationshipStorageException {
-    //return profileESConnector.search(existingIdentity, profileFilter, Relationship.Type.CONFIRMED, offset, limit);
-    return searchConnectionByFilter(existingIdentity, Type.CONFIRMED, profileFilter, offset, limit);
+    return profileESConnector.search(existingIdentity, profileFilter, Relationship.Type.CONFIRMED, offset, limit);
+    //return searchConnectionByFilter(existingIdentity, Type.CONFIRMED, profileFilter, offset, limit);
```
Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSRelationshipStorageImpl.java]: Remove comments

File: lib/src/main/java/org/exoplatform/social/addons/storage/dao/jpa/query/ProfileQueryBuilder.java [hunk 3]
```diff
@@ -100,99 +102,115 @@ public ProfileQueryBuilder withFilter(ExtendProfileFilter filter) {
         predicates.add(cb.equal(identity.get(IdentityEntity_.providerId), filter.getProviderId()));
       }
 
+      List<Identity> excludes = filter.getExcludedIdentityList();
```
Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/storage/dao/jpa/query/ProfileQueryBuilder.java]: Please add comments to explain the steps of the query build.

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java (hunk 1 above)
Comment (nttuyen) [inline lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java]: Actually the field "outgoing" contains all IDs of Identities who receive the connection request from current user. When userA sends a request connect to userB then userA will be contained on "incoming" field of userB and userB will be contained on "outgoing" field of userA.

So, If we want to search on "outgoing" of a people, we have to search on all Identities who contain his ID on their "incoming" field. That's why I reverse the "incoming" and "outgoing".

**Comment by nttuyen** [comment]
PR was updated, please review again @thomasdelhomenie.

File: lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java (hunk 1 above)
Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/search/ProfileSearchConnector.java]: ok, got it, thanks

**Comment by thomasdelhomenie** [comment]
Using ES for connections search will be done in https://github.com/exo-addons/social-rdbms/pull/7.
I close this one.
@nttuyen can you have a look at the PR https://github.com/exo-addons/social-rdbms/pull/7 to check if it includes what you have done in this PR please ?

**Pull request closed without merging**